encoding/json/escape_test.go

```diff
@@ -0,0 +1,49 @@
+package json
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestEscapeStringBytes(t *testing.T) {
+	cases := map[string]struct {
+		expected string
+		input    []byte
+	}{
+		"safeSet only": {
+			expected: `"mountainPotato"`,
+			input:    []byte("mountainPotato"),
+		},
+		"parenthesis": {
+			expected: `"foo\""`,
+			input:    []byte(`foo"`),
+		},
+		"double escape": {
+			expected: `"hello\\\\world"`,
+			input:    []byte(`hello\\world`),
+		},
+		"new line": {
+			expected: `"foo\\nbar"`,
+			input:    []byte(`foo\nbar`),
```

Comment:
This test isn't actually testing what you are are expecting. This is only testing whether `\` gets escaped as `\\`. The issue is you are using [raw string literals](https://tip.golang.org/ref/spec#String_literals) when constructing your byte slices. So you aren't actually testing that newline is getting escaped in this case. You need to be using an actual string literal to your byte slice `[]byte("foo\nbar")` which should output the expected value of `"foo\nbar"`

Reply:
fixed